### PR TITLE
New article on Performance

### PR DESCRIPTION
**Pull request by stitot**

### CODE REVIEW

File: docs/datagrid/performance.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Performance and rendering
+Row virtualization is a performance optimization technique used in DataGrid.  Instead of rendering all the rows of data at once, which can be thousands of entries, row virtualization renders only the rows that are currently visible within the user's viewport (plus a small buffer). As the user scrolls, the DataGrid dynamically renders new rows and removes those that scroll out of view.
```
Comment (jomar-honsi) [inline docs/datagrid/performance.md:2]: Personally I don't find anything wrong, but Grammarly suggests to leave out _that are_.

File: docs/datagrid/performance.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+![Illustration showing virtualization of rows](ill_virtualization.png)
+
+This technique significantly enhances DataGrid performance, leading to faster load times and efficient updates. Row virtualization reduces memory consumption, allowing the application to handle large datasets smoothly without compromising performance. Users benefit from a smoother experience with fluid scrolling and more responsive interactions when e.g. sorting rows.
```
Comment (jomar-honsi) [inline docs/datagrid/performance.md:6]: Grammarly: interactions when e.g. sorting rows-> interactions, such as sorting rows.

File: docs/datagrid/performance.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            strictHeights: true // default is false
+        }
+    }
+}
+```
+
+The `bufferSize` and `strictHeights` options can be adjusted to optimize performance and smoothness of scrolling based on your spesific use case and preferences. 
+
+### buffersize
+Defines the number of rows rendered outside the viewport (the buffer) during scrolling. A small buffer leads to faster initial rendering, and increased performance on sorting and scroll. A larger buffer means slower initial rendering, but might decrease flicker effect on fast scrolling.
```
Comment (jomar-honsi) [inline docs/datagrid/performance.md:23]: - no comman after "rendering"
- decrease _the_ flicker effect

**Review by jomar-honsi — CHANGES_REQUESTED** [review]
Looks good, just a few minor typo's etc. to fix.

File: docs/datagrid/performance.md (hunk 1 above)
Comment (sebastianbochan) [inline docs/datagrid/performance.md:2]: Double spaces before "Instead".

File: docs/datagrid/performance.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+The default `bufferSize` of 10 should be optimal in most cases.
```
Comment (sebastianbochan) [inline docs/datagrid/performance.md:25]: It would be good to link bufferSize to API docs, at least once.

File: docs/datagrid/performance.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+The `bufferSize` and `strictHeights` options can be adjusted to optimize performance and smoothness of scrolling based on your spesific use case and preferences. 
+
+### buffersize
+Defines the number of rows rendered outside the viewport (the buffer) during scrolling. A small buffer leads to faster initial rendering, and increased performance on sorting and scroll. A larger buffer means slower initial rendering, but might decrease flicker effect on fast scrolling.
+
+The default `bufferSize` of 10 should be optimal in most cases. 
+
+### strictHeights
+By default, rows adjust their height to fit all content, which can reduce performance and scrolling smoothness. Setting `strictHeights: true` ensures all rows have a uniform height, truncating multiline text with an ellipsis. This skips height calculations and boosts performance.
```
Comment (sebastianbochan) [inline docs/datagrid/performance.md:28]: It would be good to add a link to API docs.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: docs/datagrid/performance.md (hunk 5 above)
Comment (stitot) [reply docs/datagrid/performance.md:28]: @sebastianbochan: agree, but I don't know how to find correct links to API ref that is not in production. Could you help out here and "Add a suggestion"?

File: docs/datagrid/performance.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    rendering: {
+        rows: {
+            bufferSize: 3, // default is 10
+            strictHeights: true // default is false
+        }
+    }
+}
+```
+
+The `bufferSize` and `strictHeights` options can be adjusted to optimize performance and smoothness of scrolling based on your spesific use case and preferences.
```
Comment (ddragula) [inline docs/datagrid/performance.md:20]:
```suggestion
The [`bufferSize`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.RowsSettings#bufferSize) and [`strictHeights`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.RowsSettings#strictHeights) options can be adjusted to optimize performance and smoothness of scrolling based on your spesific use case and preferences. 
```

File: docs/datagrid/performance.md (hunk 4 above)
Comment (ddragula) [inline docs/datagrid/performance.md:25]:
```suggestion
The default [`bufferSize`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.RowsSettings#bufferSize) of 10 should be optimal in most cases. 
```

File: docs/datagrid/performance.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        },
+        columns: {
+            distribution: "fixed" // default is "full"
+        }
+    }
+}
+```
+These options can be used to configure how the table should be rendered.
+
+`table.className` is appended to the `<table>` element, and can be used to [style the DataGrid by CSS](https://www.highcharts.com/docs/datagrid/style-by-css).
```
Comment (ddragula) [inline docs/datagrid/performance.md:48]:
```suggestion
[`table.className`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.TableSettings#className) is appended to the `<table>` element, and can be used to [style the DataGrid by CSS](https://www.highcharts.com/docs/datagrid/style-by-css).
```

File: docs/datagrid/performance.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+`header.enabled: false` disables all [column headers](https://www.highcharts.com/docs/datagrid/header) by not rendering the `thead` element.
```
Comment (ddragula) [inline docs/datagrid/performance.md:50]:
```suggestion
[`header.enabled: false`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.HeaderSettings#enabled) disables all [column headers](https://www.highcharts.com/docs/datagrid/header) by not rendering the `thead` element.
```

File: docs/datagrid/performance.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+`columns.distribution` is used to configure initial column widths.
```
Comment (ddragula) [inline docs/datagrid/performance.md:52]:
```suggestion
[`columns.distribution`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.ColumnsSettings#distribution) is used to configure initial column widths.
```

File: docs/datagrid/performance.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+`columns.distribution:"full"` renders a full width (`width: 100%;`) responsive table with evenly distributed column widths. `columns.distribution:"fixed"` renders a table were columns have a fixed width in pixels. Should be set to support your spesific use case.
```
Comment (ddragula) [inline docs/datagrid/performance.md:54]:
```suggestion
[`columns.distribution: "full"`](https://api.highcharts.com/dashboards/#interfaces/DataGrid_Options.ColumnsSettings#distribution) renders a full width (`width: 100%;`) responsive table with evenly distributed column widths. `columns.distribution:"fixed"` renders a table were columns have a fixed width in pixels. Should be set to support your spesific use case.
```

**Review by ddragula — APPROVED** [review]
Approved! + Suggestions with links to API Docs below. 😄

**Review by sebastianbochan — APPROVED** [review]

**Review by ddragula — APPROVED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**